src/ReverseProxy/Model/HttpContextFeaturesExtensions.cs

```diff
@@ -60,4 +60,22 @@ public static void ReassignProxyRequest(this HttpContext context, ClusterState c
         };
         context.Features.Set<IReverseProxyFeature>(newFeature);
     }
+
+    // ReassignProxyRequest overload to also replace the route when updating IReverseProxyFeature
+    // See discussion in https://github.com/microsoft/reverse-proxy/discussions/1749
+    // and https://github.com/microsoft/reverse-proxy/issues/1752
+    public static void ReassignProxyRequest(this HttpContext context, ClusterState cluster, RouteModel route)
```

Comment:
```suggestion
    public static void ReassignProxyRequest(this HttpContext context, RouteModel route, ClusterState cluster)
```
We generally use the "routes, clusters" order.